common/src/main/java/org/apache/falcon/entity/ClusterHelper.java

```diff
@@ -211,4 +219,55 @@ public static String getEmptyDir(Cluster cluster) {
         return getStorageUrl(cluster) + getLocation(cluster, ClusterLocationType.STAGING).getPath()
                 + "/" + EMPTY_DIR_NAME;
     }
+
+    public static boolean matchInterface(final Cluster oldEntity, final Cluster newEntity,
+                                         final Interfacetype interfaceType) {
+        Interface oldInterface = getInterface(oldEntity, interfaceType);
+        Interface newInterface = getInterface(newEntity, interfaceType);
+        String oldEndpoint = (oldInterface == null) ? null : oldInterface.getEndpoint();
+        String newEndpoint = (newInterface == null) ? null : newInterface.getEndpoint();
+        LOG.debug("Verifying if Interfaces match for cluster {} : Old - {}, New - {}",
+                interfaceType.name(), oldEndpoint, newEndpoint);
+        return StringUtils.isBlank(oldEndpoint) && StringUtils.isBlank(newEndpoint)
+                || StringUtils.isNotBlank(oldEndpoint) && oldEndpoint.equalsIgnoreCase(newEndpoint);
+    }
+
+    public static boolean matchLocations(final Cluster oldEntity, final Cluster newEntity,
+                                         final ClusterLocationType locationType) {
+        Location oldLocation = getLocation(oldEntity, locationType);
+        Location newLocation = getLocation(newEntity, locationType);
+        String oldLocationPath = (oldLocation == null) ? null : oldLocation.getPath();
+        String newLocationPath = (newLocation == null) ? null : newLocation.getPath();
+        LOG.debug("Verifying if Locations match {} : Old - {}, New - {}",
+                locationType.name(), oldLocationPath, newLocationPath);
+        return  StringUtils.isBlank(oldLocationPath) && StringUtils.isBlank(newLocationPath)
+                || StringUtils.isNotBlank(oldLocationPath) && oldLocationPath.equalsIgnoreCase(newLocationPath);
+    }
+
+    public static boolean matchProperties(final Cluster oldEntity, final Cluster newEntity) {
+        Map<String, String> oldProps = new HashMap<String, String>();
+        Map<String, String> newProps = new HashMap<String, String>();
+
+        if (oldEntity.getProperties() != null) {
+            for (Property prop : oldEntity.getProperties().getProperties()) {
+                oldProps.put(prop.getName(), prop.getValue());
+            }
+        }
+        if (newEntity.getProperties() != null) {
+            for (Property prop : newEntity.getProperties().getProperties()) {
+                newProps.put(prop.getName(), prop.getValue());
+            }
+        }
+
```

Comment:
Better to add a function to return properties of a cluster, as what you did for location and interface.